app/sanitizers/rendered_markdown_scrubber.rb

```diff
@@ -18,9 +18,32 @@ def allowed_node?(node)
     @tags.include?(node.name) || valid_codeblock_div?(node)
   end
 
+  # Overrides scrub_attributes in
+  # https://github.com/rails/rails-html-sanitizer/blob/master/lib/rails/html/scrubbers.rb
+  def scrub_attributes(node)
+    # We only want to call super if we aren't in a codeblock
+    # because the `class` attribute will be stripped
+    if inside_codeblock?(node)
+      node.attribute_nodes.each do |attr|
+        scrub_attribute(node, attr)
+      end
+
+      scrub_css_attribute(node)
+    else
+      super
+    end
+  end
+
   private
 
+  def inside_codeblock?(node)
+    node.attributes["class"]&.value&.include?("highlight") ||
+      (node.name == "span" && node.ancestors.first.name == "code")
+  end
+
   def valid_codeblock_div?(node)
+    return false if node.nil?
```

Comment:
Is node being `nil` something that happens due to this PR or was it something that could feasibly happen before and you're just patching it?

Reply:
It could have feasibly happened before, but this doesn't strictly need to be here - it got left in from when I was taking another approach. I'll remove it in another PR.